src/openms/source/SIMULATION/MSSim.cpp

```diff
@@ -528,12 +528,10 @@ namespace OpenMS
     proteins.insert(proteins.end(), prot_ids.begin(), prot_ids.end());
 
     // peptide IDs
+    peptides.reserve(fmap.size());
     for (FeatureMap::ConstIterator it = fmap.begin(); it != fmap.end(); ++it)
     {
-      const Feature& f = *it;
-      const vector<PeptideIdentification>& pep_ids = f.getPeptideIdentifications();
-      peptides.reserve(peptides.size() + pep_ids.size());
-      peptides.insert(peptides.end(), pep_ids.begin(), pep_ids.end());
+      peptides.push_back(it->getPeptideIdentifications()[0]);
```

Comment:
I think this feature map is the ground truth. A feature here corresponds to exactly one peptide, which has exactly one identity. `f.getPeptideIdentifications()[0]` is used everywhere in the simulator code:

`$ find . -name '*Sim*cpp' | grep -v test | while read i; do grep -l 'getPeptideIdentifications()\[0\]' $i; done
./src/openms/source/SIMULATION/DetectabilitySimulation.cpp
./src/openms/source/SIMULATION/DigestSimulation.cpp
./src/openms/source/SIMULATION/IonizationSimulation.cpp
./src/openms/source/SIMULATION/MSSim.cpp
./src/openms/source/SIMULATION/RawMSSignalSimulation.cpp
./src/openms/source/SIMULATION/RawTandemMSSignalSimulation.cpp
./src/openms/source/SIMULATION/RTSimulation.cpp`

`
$ find . -name '*Sim*cpp' | grep -v test | while read i; do grep 'getPeptideIdentifications()\[0\]' $i; done
      peptides_vector[i] = features[i].getPeptideIdentifications()[0].getHits()[0].getSequence().toUnmodifiedString();
        std::set<String> protein_accessions = generated_features[*dp_it].getPeptideIdentifications()[0].getHits()[0].extractProteinAccessions();
        UInt basic_residues_c = countIonizedResidues_(features[index].getPeptideIdentifications()[0].getHits()[0].getSequence());
    EmpiricalFormula feature_ef = f.getPeptideIdentifications()[0].getHits()[0].getSequence().getFormula();
    std::vector<PeptideHit> hits = f.getPeptideIdentifications()[0].getHits();
    f.getPeptideIdentifications()[0].setHits(hits);
             << " Pep: " << feat->getPeptideIdentifications()[0].getHits()[0].getSequence().toString() << endl
        for (vector<String>::const_iterator it = (*feat).getPeptideIdentifications()[0].getHits()[0].getProteinAccessions().begin();
             it != (*feat).getPeptideIdentifications()[0].getHits()[0].getProteinAccessions().end();
      PeptideIdentification& pi = f.getPeptideIdentifications()[0];
      peptides.push_back(it->getPeptideIdentifications()[0]);
    EmpiricalFormula ef = active_feature.getPeptideIdentifications()[0].getHits()[0].getSequence().getFormula();
      ef = EmpiricalFormula(active_feature.getPeptideIdentifications()[0].getHits()[0].getSequence().getFormula());
      AASequence seq = features[i_f].getPeptideIdentifications()[0].getHits()[0].getSequence();
        feature_seq.push_back(features[i_f].getPeptideIdentifications()[0].getHits()[0].getSequence().toString() + ":" + features[i_f].getCharge());
        AASequence seq = features[ids[id]].getPeptideIdentifications()[0].getHits()[0].getSequence();
        peptides_aa_vector[i] = features[i].getPeptideIdentifications()[0].getHits()[0].getSequence();
        deleted_features.push_back(features[i].getPeptideIdentifications()[0].getHits()[0].getSequence().toUnmodifiedString() + " [" +
      String seq = features[i].getPeptideIdentifications()[0].getHits()[0].getSequence().toUnmodifiedString();
      features[i].getPeptideIdentifications()[0].getHits()[0].getSequence().getAAFrequencies(frequency_table);
      double mass = features[i].getPeptideIdentifications()[0].getHits()[0].getSequence().getFormula().getAverageWeight();`